src/schema-generation/flex-search-generator.ts

```diff
@@ -0,0 +1,186 @@
+            };
+
+
+        return {
+            ...schemaField,
+            args: {
+                ...schemaField.args,
+                ...newArgs
+            },
+            resolve: (sourceNode, args, info) => {
+                const itemVariable = new VariableQueryNode(decapitalize(rootEntityType.name));
+                const flexFilterNode = this.buildFlexSearchFilterNode(args, flexSearchType, itemVariable, rootEntityType, info);
+                return new FlexSearchQueryNode({
+                    rootEntityType: rootEntityType,
+                    flexFilterNode: flexFilterNode,
+                    itemVariable: itemVariable
+                });
+            }
+
+
+        };
+
+
+    };
+
+    private buildFlexSearchFilterNode(args: { [p: string]: any }, filterType: FlexSearchFilterObjectType, itemVariable: VariableQueryNode, rootEntityType: RootEntityType, info: QueryNodeResolveInfo) {
+        const filterValue = args[FLEX_SEARCH_FILTER_ARG] || {};
+        const expression = rootEntityType.hasIncludedInSearchFields ? args[FLEX_SEARCH_EXPRESSION_ARG] as string : undefined;
+        const filterNode = simplifyBooleans(filterType.getFilterNode(itemVariable, filterValue, [], info));
+        const searchFilterNode = simplifyBooleans(this.buildFlexSearchSearchFilterNode(rootEntityType, itemVariable, expression));
+        if (searchFilterNode === ConstBoolQueryNode.TRUE) {
+            return filterNode;
+        } else {
+            return simplifyBooleans(new BinaryOperationQueryNode(filterNode, BinaryOperator.AND, searchFilterNode));
+        }
+
+    }
+
+    private getPreExecQueryNode(rootEntityType: RootEntityType, args: { [p: string]: any }, context: QueryNodeResolveInfo): QueryNode {
+        const itemVariable = new VariableQueryNode(decapitalize(rootEntityType.name));
+        const flexSearchType = this.flexSearchTypeGenerator.generate(rootEntityType, false);
+        const flexFilterNode = this.buildFlexSearchFilterNode(args, flexSearchType, itemVariable, rootEntityType, context);
+        return new BinaryOperationQueryNode(
+            new CountQueryNode(
+                new FlexSearchQueryNode({
+                    rootEntityType: rootEntityType,
+                    flexFilterNode: flexFilterNode,
+                    itemVariable: itemVariable
+                })),
+            BinaryOperator.GREATER_THAN,
+            new LiteralQueryNode(context.flexSearchMaxFilterableAmountOverride ? context.flexSearchMaxFilterableAmountOverride : DEFAULT_ARANGOSEARCH_MAX_FILTERABLE_AMOUNT));
+    }
+
+    private augmentWithCondition(schemaField: QueryNodeField, rootEntityType: RootEntityType): QueryNodeField {
+        return {
+            ...schemaField,
+            transform: (sourceNode, args, context) => {
+                const assertionVariable = new VariableQueryNode();
+                if (args[FILTER_ARG] || args[ORDER_BY_ARG]) {
```

Comment:
could you add a comment on how this works? It took me a bit to realize that `sourceNode` already has the list augmentation applied, so the filters are included, and `this.getPreExecQueryNode` creates the flexsearch node without the listaugmentation.